### DISCUSSION TITLE
Purpose of .lim for eeg_artif_step

### DISCUSSION THREAD

**Discussion by stonekate** [bnicenboim/eeguana#227] [Q&A] [answered]
Hey Bruno,

Someone has asked me and I'm not actually sure of the answer: What is the reason for specifying a window for eeg_artif_step? The function seems to be testing whether the voltage changes more than some threshold between one datapoint and the next (same as step functions in other preprocessing packages), so why does one need to specify a time window? From your code it does look like the window is being used, is it maybe just to minimise compute or something? 

TIA!

**Comment by bnicenboim** [comment]
ha, I don't remember! It does something and I had to have it, but I don't remember why.
My plan is to create a vignette with some square and sin waves and show what happens with all these functions in all sort of situations...
My computer is constantly crashing, and I couldn't even make the unit test of the package work.... Once I manage to do that, I'll do a vignette (also for myself) :)

**Reply by stonekate** [reply]
lol ok! That sounds super helpful, even as a teaching tool :D (but mainly for myself learning)

**Reply by stonekate** [reply]
ah, could it be that the time window is for marking the data around the step? as in, it's not relevant for detecting the step, just for marking the events table?

**Reply by bnicenboim** [reply, marked as answer]
ok, a bit late, but I remembered why. It's taking an average of a window, you don't just want a difference between two datapoints, you want a proper step. It's described in chapter 6 of Luck (2014)

**Reply by stonekate** [reply]
Ah thanks! I read it, makes sense. And thanks for the reference, I should move on from his 2005 book.
